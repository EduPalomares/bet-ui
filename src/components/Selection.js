import React, { Component } from 'react';
import { Consumer } from './Provider';

export default class Selection extends Component {

	constructor(props) {
		super(props);
		
		this.state = {
			isSelected: false,
		};

		this.betId = this.props.selection.id;
		this.handleStatusBet = this.handleStatusBet.bind(this);
		this.thisSelect = this.thisSelect.bind(this);

	}
  
	thisSelect() {
		this.setState((prevState, props) => ({
			isSelected: true
		}));
	}
	
	handleStatusBet(betsIds) {
		let selected = (betsIds.indexOf(this.props.selection.id)>=0) ? true : false;
		if (this.state.isSelected!==selected) {
			this.setState((prevState, props) => ({
				isSelected: selected
			}));
		}
	}
	
	render() {

		return (
			<Consumer>
				{
					context => {
					
						this.handleStatusBet(context.betsIds);
						
						return (
							<div className={(this.state.isSelected) ? "selection selected" : "selection"} onClick={	() => { this.thisSelect(); context.addBet(this.betId) }}>
								<div className="name">{this.props.selection.name}</div>
								<div className="price">{this.props.selection.price}</div>
							</div>
						)
					}
				}
			</Consumer>
		);
		
	}
}



