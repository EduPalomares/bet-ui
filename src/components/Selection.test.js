import React from 'react';
import Selection from './Selection';

const props = {
	selection: {"id": "SEL_1", "name": "Real Madrid", "price": 1.23},
	betsIds: [],
	context: {
		betsIds: []
	}
};

const ConsumerSelection = shallow(<Selection {...props} />).props().children;
const wrapper = shallow(<ConsumerSelection {...props} />);

test('render match snapshot', () => {
	expect(wrapper).toMatchSnapshot();
});

test('contains name & price', () => {
	expect(wrapper.find('.name').length).toBe(1);
	expect(wrapper.find('.price').length).toBe(1);
});