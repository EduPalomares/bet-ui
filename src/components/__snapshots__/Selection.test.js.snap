// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    betsIds={Array []}
    context={
        Object {
            "betsIds": Array [],
          }
    }
    selection={
        Object {
            "id": "SEL_1",
            "name": "Real Madrid",
            "price": 1.23,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="name"
>
          Real Madrid
</div>,
        <div
          className="price"
>
          1.23
</div>,
      ],
      "className": "selection",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Real Madrid",
          "className": "name",
        },
        "ref": null,
        "rendered": "Real Madrid",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 1.23,
          "className": "price",
        },
        "ref": null,
        "rendered": 1.23,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="name"
>
            Real Madrid
</div>,
          <div
            className="price"
>
            1.23
</div>,
        ],
        "className": "selection",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Real Madrid",
            "className": "name",
          },
          "ref": null,
          "rendered": "Real Madrid",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 1.23,
            "className": "price",
          },
          "ref": null,
          "rendered": 1.23,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
